contracts/vaults/RibbonThetaVault.sol

```diff
@@ -434,6 +435,40 @@ contract RibbonThetaVault is OptionsVaultStorage {
         transferAsset(msg.sender, amount);
     }
 
+    /**
+     * @notice Initiates a withdrawal that can be processed once the round completes
+     * @param shares is the number of shares to withdraw
+     */
+    function initiateWithdraw(uint128 shares) external nonReentrant {
+        require(shares > 0, "!shares");
+
+        // This caches the `round` variable used in shareBalances
+        uint16 currentRound = round;
+        Vault.Withdrawal memory withdrawal = withdrawals[msg.sender];
+
+        require(!withdrawal.initiated, "Existing withdraw");
```

Comment:
so we are constraining it to only one withdrawal per round?

ex:

initiateWithdraw(half of my shares)
10 min later
initiateWithdraw(quarter of my shares)

Reply:
Yep, we're limiting only 1 withdrawal per round. This is really for simplicity